src/um/src/um_ovsdb.c

```diff
@@ -329,6 +329,11 @@ static void callback_AWLAN_Node(
                 if (strlen(awlan_node->firmware_url) > 0)
                 {
                     if(upg_url == NULL || strncmp(upg_url, awlan_node->firmware_url, sizeof(awlan_node->firmware_url))){
+                        if(upg_url)
+                        {
+                           free(upg_url);
+                           upg_url = NULL;
+                        }
```

Comment:
```suggestion
                        if (upg_url) free(upg_url);
```

During an internal review, it was noticed that setting the `upg_url` to `NULL` is not needed.

Strictly speaking, neither is checking for `upg_url` not being `NULL`, because `free()` is guaranteed to be NO-OP for NULL, but it doesn't hurt either and it adds some clarity here when reading the code.